src/environment.rs

```diff
@@ -357,6 +357,13 @@ fn locked_dependency_satisfies(
             (Some(_), None) => return false,
             _ => {}
         }
+
+        // Test if the locked channel is equal to the channel that is requested.
+        if let Some(channel) = &spec.channel {
+            if !conda.url.as_ref().contains(channel) {
```

Comment:
I think in the future we should have a proper channel type and compare the real "channels" instead of `contains` (I am assuming this would find `conda-forge` in the full URL, right?

This is not required for this PR IMO, but would be a good future enhancement.

Reply:
Yeah this seems very fragile. This will also work for pytorch with the pytorch-nightly channel for instance which might be very confusing. Maybe just compare the base_url since I think thats whats usually stored in the channel field.

Reply:
Yeah I completely agree. Just don't know how to fix it correctly. I guess we need to do some normalization after the parsing. IIRC, I left a todo in rattler about this.

Reply:
I made the check much more precise.